File: vue-vben-admin/apps/web-naive/src/api/enum/ImageFieldTypeEnum.ts
export enum ImageFieldTypeEnum {
  AbsoluteHref = 'absoluteHref',    // 完整链接
  AbsolutePath = 'absolutePath',    // 绝对路径
  BarCode = 'barCode',    // 条形码
  QrCode = 'qrCode',    // 二维码
}

export const ImageFieldTypeEnumOption = [
  { label: '完整链接', value: 'absoluteHref' },
  { label: '绝对路径', value: 'absolutePath' },
  { label: '条形码', value: 'barCode' },
  { label: '二维码', value: 'qrCode' },
];

export function getImageFieldTypeLabel(value: string): string | undefined {
  const item = ImageFieldTypeEnumOption.find(x => x.value === value);
  return item?.label;
}

File: src/SimpleApp/Simple.Interfaces/Enums/ImageFieldTypeEnum.cs
using System.ComponentModel;
using Simple.Utils.Attributes;

namespace Simple.Interfaces.Enums;

[GenerateTypeScript]
public static class ImageFieldTypeEnum
{
    /// <summary>
    /// 绝对路径
    /// </summary>
    [Description("绝对路径")]
    public static string AbsolutePath = "absolutePath";

    /// <summary>
    /// 绝对链接
    /// </summary>
    [Description("完整链接")]
    public static string AbsoluteHref = "absoluteHref";
    
    /// <summary>
    /// 条形码
    /// </summary>
    [Description("条形码")]
    public static string BarCode = "barCode";

    /// <summary>
    /// 二维码
    /// </summary>
    [Description("二维码")]
    public const string QrCode = "qrCode";
}